<?xml version="1.0" encoding="UTF-8"?>
<beans xmlns="http://www.springframework.org/schema/beans"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xsi:schemaLocation="http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans-2.0.xsd">

    <!-- $Revision$ -->
    <!-- $URL$ -->
    <!-- $Date$ -->
    <!-- $Author$ -->
    
    <import resource="classpath:scenarios/common/remotingtests-soap-protocol-config.xml"/>
    
    <bean id="soapCalculatorExporter" class="ch.elca.el4j.services.remoting.RemotingServiceExporter">
        <property name="remoteProtocol">
            <ref bean="soapProtocol" />
        </property>
        <property name="serviceInterface">
            <value>ch.elca.el4j.tests.remoting.service.Calculator</value>
        </property>
        <property name="serviceName">
            <value>Calculator.Soap.Remotingtests</value>
        </property>
        <property name="service">
            <idref bean="soapCalculatorImpl"/>
        </property>
    </bean>
    
    <bean id="soapCalculatorImpl" class="ch.elca.el4j.tests.remoting.service.impl.CalculatorImpl" />

    <bean id="soapExceptionThrowerExporter" class="ch.elca.el4j.services.remoting.RemotingServiceExporter">
        <property name="remoteProtocol">
            <ref bean="soapProtocolForSpecificSoapConformExceptions" />
        </property>
        <property name="serviceInterface">
            <value>ch.elca.el4j.tests.remoting.service.soap.ExceptionThrower</value>
        </property>
        <property name="service">
            <idref bean="soapExceptionThrowerImpl"/>
        </property>
    </bean>
    
    <bean id="soapExceptionThrowerImpl" class="ch.elca.el4j.tests.remoting.service.soap.impl.ExceptionThrowerImpl" />
</beans>
